BLL/Services/PaymentCapture.cs

```diff
@@ -5,48 +5,44 @@
 using BLL.Helpers;
 using BLL.Helpers.Interfaces;
 using BLL.Helpers.Mapping.Interfaces;
+using BLL.Helpers.Queries.Interfaces;
 using BLL.Models;
 using BLL.Services.Interfaces;
-using DAL.Constants;
 using DAL.Repositories.Interfaces;
 using DAL.DBModels;
 using Stripe;
 
 namespace BLL.Services
 {
-    public class PaymentCapture : IPaymentExecute
+    public class PaymentCapture : PaymentExecuteBase
     {
-        private readonly IPaymentRepository _paymentRepository;
-        private readonly IMappingProvider _mappingProvider;
-        private readonly IRetryHelper _retryHelper;
-
-        public PaymentCapture(IPaymentRepository paymentRepository, IMappingProvider mappingProvider, IRetryHelper retryHelper)
+        private readonly ITransactionQueryCreator _queryCreator;
+        public PaymentCapture(ITransactionRepository paymentRepository, IMappingProvider mappingProvider, IRetryHelper retryHelper, ITransactionQueryCreator queryCreator)
+            : base(paymentRepository, mappingProvider, retryHelper)
         {
-            _paymentRepository = paymentRepository;
-            _mappingProvider = mappingProvider;
-            _retryHelper = retryHelper;
+            _queryCreator = queryCreator;
         }
 
-        public async Task<IEnumerable<TransactionDTO>> Execute(PaymentModel payment)
+        public override async Task<IEnumerable<TransactionDTO>> Execute(PaymentModel payment)
         {
-            var customer = await _paymentRepository.GetLastTransaction(payment.OrderId);
+            var customer = (await PaymentRepository.GetTransactions(await _queryCreator.GetTransactionForCapture(payment.OrderId, PaymentServiceConstants.PaymentType.Auth))).LastOrDefault();
```

Comment:
Possible null reference exception in case customer is not found

Reply:
I will add new custom exception